protostar/starknet/address.py

```diff
@@ -0,0 +1,42 @@
+from typing import Union
+
+from typing_extensions import Self
+
+from protostar.protostar_exception import ProtostarException
+
+
+class Address:
+    @classmethod
+    def from_user_input(cls, raw_address: Union[str, int]) -> Self:
+        value = (
+            raw_address if isinstance(raw_address, int) else int(raw_address, base=0)
+        )
+        if value < 0:
+            raise AddressValidationError(raw_address)
+        return cls(value)
+
+    def __init__(self, value: int) -> None:
+        self._value = value
+
+    def as_int(self) -> int:
+        return self._value
+
+    def as_str(self) -> str:
```

Comment:
why bother with a separate method which does that?

Reply:
yep, use `__str__` and `__int__`

Reply:
Replaced with dunder methods.

I created explicit methods because it's not obvious what representation are supported without knowing internal implementation.